Mapping for maxAccessEgressDurationPerMode in Transmodel API

Bartosz-Kruba opened the pull request:
### Summary

This PR add mapping for maxAccessEgressDurationPerMode to Transmodel API.

### Issue
N/A

### Unit tests
N/A

### Documentation
Documented in GraphQL schema

hannesj commented on src/ext/graphql/transmodelapi/schema.graphql:
```diff
@@ -1840,6 +1842,20 @@ input Location {
 }
 
+"Maximum duration for access/egress for street searches per respective mode."
+input MaxAccessEgressInput {
+  bike: Duration
+  bikeRental: Duration
+  bikeToPark: Duration
+  car: Duration
+  carPickup: Duration
+  carRental: Duration
+  carToPark: Duration
+  flexible: Duration
+  scooterRental: Duration
+  walk: Duration
+}
```
I think a list of streetmode - duration pairs would be a better format for these

Bartosz-Kruba replied:
I've changed it now

hannesj commented on src/ext/graphql/transmodelapi/schema.graphql:
```diff
@@ -1768,6 +1770,12 @@ scalar Long
 scalar Time
 
+"A combination of street mode and corresponding duration"
+input DurationPerModeInput {
```
We have a similar type, `TransportModeSlack`, so a equivalent name for the type could be `StreetModeDuration`

hannesj commented on src/main/java/org/opentripplanner/routing/api/request/preference/StreetPreferences.java:
```diff
@@ -179,6 +179,13 @@ public Builder withIntersectionTraversalModel(IntersectionTraversalModel model)
+    public Builder withMaxAccessEgressDuration(
+      DurationForEnum<StreetMode> maxAccessEgressDuration
+    ) {
+      this.maxAccessEgressDuration = maxAccessEgressDuration;
```
I think we need to merge this with `this.maxAccessEgressDuration`, not just overwrite

hannesj commented on src/ext/java/org/opentripplanner/ext/transmodelapi/mapping/PreferencesMapper.java:
```diff
@@ -24,6 +31,24 @@ static void mapPreferences(
+
+    if (GqlUtil.hasArgument(environment, "maxAccessEgressDurationForMode")) {
+      preferences.withStreet(street -> {
+        var maxAccessEgressDuration = StreetPreferences.DEFAULT.maxAccessEgressDuration().copyOf();
```
We should not use `StreetPreferences.DEFAULT` here, it can be overridden in config. Instead we should just pass in the values from the user into `withMaxAccessEgressDuration`. Perhaps we can modify it, so that we call the function once per mode in the array.

Bartosz-Kruba commented:
@hannesj I've change the code according to your feedback. Renamed input type to `StreetModeDuration` and changed mapping code so that we do not override durations for config if mode is not specified. I should be ready for merge now.

hannesj commented on src/ext/java/org/opentripplanner/ext/transmodelapi/mapping/PreferencesMapper.java:
```diff
@@ -24,6 +31,20 @@ static void mapPreferences(
+
+    if (GqlUtil.hasArgument(environment, "maxAccessEgressDurationForMode")) {
+      preferences.withStreet(street -> {
+        for (var entry : (List<Map<String, ?>>) environment.getArgument(
+          "maxAccessEgressDurationForMode"
+        )) {
```
It is usually cleaner to be explicit with the type, when there is a need to cast
```suggestion
        for (List<Map<String, ?>> entry : environment.getArgument(
          "maxAccessEgressDurationForMode"
        )) {
```

hannesj reviewed (approved):
Only a minor suggestion
